tests/test_log_event_detector.py

```diff
@@ -76,8 +79,13 @@
         ),
     ],
 )
-def test_log_parser_single_event(sample_pattern, logs, expected_event_count):
-    parser = LogParser(event_patterns=[sample_pattern])
-    parser.parse(logs)
-    detected_events = parser.get_detected_events()
+def test_log_parser_single_event(
+    sample_pattern: EventPattern, logs: List[Log], expected_event_count: int
+):
+    parser = EventDetector(event_patterns=[sample_pattern])
+    detected_events = parser.detect_events(logs)
     assert len(detected_events) == expected_event_count
```

Comment:
```suggestion

    assert len(detected_events) == expected_event_count
```

In order for it to be in a "block" near the next asserts